tests/scripts/all.sh

```diff
@@ -1456,9 +1456,9 @@ component_test_tls1_2_ecjpake_compatibility() {
     make -C programs test/udp_proxy test/query_compile_time_config
 
     msg "test: server w/o USE_PSA - client w/ USE_PSA"
-    P_SRV=../s2_no_use_psa tests/ssl-opt.sh -f ECJPAKE
+    P_SRV=../s2_no_use_psa tests/ssl-opt.sh -f ECJPAKE -e ECJPAKE_OPAQUE_PW
```

Comment:
If the server wasn't compiled with `USE_PSA` it probably shouldn't be able to run `ECJPAKE_OPAQUE_PW` tests, like `opaque password client+server` or `opaque password server only`.
I think that this is a problem with current implementation of `ecjpake_pw_opaque` option in ssl programs: they silently ignore the request if `USE_PSA` isn't on. 
I think that `ecjpake_pw_opaque` should be available only if `USE_PSA` is on, and that these `all.sh` tests should filter, respectively:

`ECJPAKE_OPAQUE_PW: working, TLS, opaque password client only` for line 1459, where only the client has `USE_PSA` on;
`ECJPAKE_OPAQUE_PW: working, TLS, opaque password server only` for line 1461, where only the server has `USA_PSA` on.

Reply:
In fact we have to run four tests, not two:
`PSA client, clear password <=> non-PSA server, clear password`
`PSA client, opaque password <=> non-PSA server, clear password`

`non-PSA client, clear password <=> PSA server, clear password`
`non-PSA client, clear password <=> PSA server, opaque password`

And I think that this comatibility component in `all.sh` should call exactly these four tests by hand and nothing more.
I also think that `ssl-server2` and `ssl-client2` compiled without `USE_PSA` should not accept `ecjpake_pw_opaque` and error out if it's passed, that's why we can't just have one prefix like `ECJPAKE_OPAQUE_PW` to run these tests.

Following current names of the tests it should be something like this:
```suggestion
    msg "test: server w/o USE_PSA - client w/ USE_PSA, text password"
    P_SRV=../s2_no_use_psa tests/ssl-opt.sh -f "ECJPAKE: working, TLS"
    msg "test: server w/o USE_PSA - client w/ USE_PSA, opaque password"
    P_SRV=../s2_no_use_psa tests/ssl-opt.sh -f "ECJPAKE_OPAQUE_PW: working, TLS, opaque password client only"
    
    msg "test: client w/o USE_PSA - server w/ USE_PSA, text password"
    P_CLI=../c2_no_use_psa tests/ssl-opt.sh -f "ECJPAKE: working, TLS"
    msg "test: client w/o USE_PSA - server w/ USE_PSA, opaque password"
    P_CLI=../c2_no_use_psa tests/ssl-opt.sh -f "ECJPAKE_OPAQUE_PW: working, TLS, opaque password server only"    
```
The way `query_compile_time_config` works with current order of compilation inside this test - it will always think that `USE_PSA_CRYPTO` is on, since that's the config it was compiled with before running any `ssl-opt.sh` tests.
Does this make more sense now?

Reply:
Ok, this is crystal clear now! You're right, I was basically missing 2 test conditions here.
For this part:

> I also think that ssl-server2 and ssl-client2 compiled without USE_PSA should not accept ecjpake_pw_opaque and error out if it's passed

this should be fixed in [this](https://github.com/Mbed-TLS/mbedtls/pull/6680/commits/d6feb208694f33b6e612351ff02c7b061335cdef) commit.

Whereas for this one:

> In fact we have to run four tests, not two:

you can refer to [this](https://github.com/Mbed-TLS/mbedtls/pull/6680/commits/70e029006d6907cbc528664660e30a591464ed7e) commit